service/src/intTest/java/uk/nhs/adaptors/gp2gp/ehr/TimeToLiveTest.java

```diff
@@ -0,0 +1,68 @@
+package uk.nhs.adaptors.gp2gp.ehr;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import uk.nhs.adaptors.gp2gp.common.mongo.MongoClientConfiguration;
+import uk.nhs.adaptors.gp2gp.common.mongo.ttl.MongoTtlCreator;
+import uk.nhs.adaptors.gp2gp.repositories.EhrExtractStatusRepository;
+import uk.nhs.adaptors.gp2gp.testcontainers.ActiveMQExtension;
+import uk.nhs.adaptors.gp2gp.testcontainers.MongoDBExtension;
+
+import org.assertj.core.api.Assertions;
+import org.awaitility.Durations;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.index.IndexInfo;
+import org.springframework.data.mongodb.core.index.IndexOperations;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@ExtendWith({ SpringExtension.class, MongoDBExtension.class, ActiveMQExtension.class})
+@SpringBootTest
+@DirtiesContext
+public class TimeToLiveTest {
+    private static final int MAX_AWAIT_IN_SECONDS = 90;
+
+    @Autowired
+    private MongoTemplate mongoTemplate;
+    @Autowired
+    private MongoClientConfiguration mongoClientConfiguration;
+    @Autowired
+    private EhrExtractStatusRepository ehrExtractStatusRepository;
+
+    @Test
+    public void When_ApplicationStarts_Expect_TtlIndexExistsForEhrExtractStatusWithValueFromConfiguration() {
+        IndexOperations indexOperations = mongoTemplate.indexOps(EhrExtractStatus.class);
+        Assertions.assertThat(timeToLiveIndexExists(indexOperations)).isTrue();
+    }
+
+    @Test
+    @Disabled("Long running test that depends on external TTL config, enable when needed")
+    public void When_TimeToLiveHasPassedInEhrExtractStatusRepository_Expect_DocumentRemoved() {
```

Comment:
This test fails when enabled.

Reply:
If we make the default in src/intTest/resources/application.yml much shorter (a few seconds):
1) could we have this test always run without our integration tests taking way too long to finish
2) do the other repo intTest still run reliably?

Reply:
Tests were running and passing with the config set to 2 and 3 seconds, left disabled as tests fail when MAX_AWAIT_IN_SECONDS is set to less than 40 seconds, left disabled as discussed.